### PR TITLE
Update to SimpleSAMLphp 1.17

### PR DESCRIPTION
**Pull request by briskt**
- use SSP namespaces
- update local copies of SSP library file `www_saml2_idp_SSOService.php `
- update overrides of SSP library files `SP.php` and `SAML2.php`
- change functional test search text strings
- remove test-generated files previously committed

### CODE REVIEW

File: development/idp2-local/config/config.php [hunk 1]
```diff
@@ -141,17 +141,17 @@
      *
      * define the minimum log level to log
      *		SimpleSAML\Logger::ERR		No statistics, only errors
```
Comment (Baggerone) [inline development/idp2-local/config/config.php]:
```suggestion
      *		\SimpleSAML\Logger::ERR		No statistics, only errors
```

File: lib/IdP/SAML2.php [hunk 2]
```diff
@@ -306,42 +319,54 @@ public static function receiveAuthnRequest(SimpleSAML_IdP $idp)
                 $nameIDFormat = null;
             }
 
+            if (isset($_REQUEST['ConsumerURL'])) {
+                $consumerURL = (string)$_REQUEST['ConsumerURL'];
+            } elseif (isset($_REQUEST['shire'])) {
+                $consumerURL = (string)$_REQUEST['shire'];
```
Comment (forevermatt) [inline lib/IdP/SAML2.php:325]: I didn't know [the Shire](https://en.wikipedia.org/wiki/Shire_(Middle-earth)) used SAML... 😉

Comment (briskt) [reply lib/IdP/SAML2.php:325]: Haha, now you know...

**Review by forevermatt — APPROVED** [review]

**Review by fillup — APPROVED** [review]

**Review by Baggerone — APPROVED** [review]
Tiny low-importance suggestion in a comment.

**Pull request merged**